explainer.md

```diff
@@ -0,0 +1,208 @@
+
+```js
+const buttons = [
+   {position: [1, 0, 0, 1], radius: 0.1, pressed: false,
+    onpress: function() { ... }, onrelease: function() { ... }},
+   // ...  
+];
+
+function checkInteraction(button, inputSource, frame, renderer) {
+   let tip = frame.getPose(inputSource.hand[XRJoint.INDEX_PHALANGE_TIP], renderer.referenceSpace);
+   let distance = calculateDistance(tip.transform.position, button.position);
+   if (distance < button.radius) {
+      if (!button.pressed) {
+         button.pressed = true;
+         button.onpress();
+      }
+   } else {
+      if (button.pressed) {
+         button.pressed = false;
+         button.onrelease();
+      }
+   }
+}
+
+function onFrame(frame, renderer) {
+   // ...
+   for (button of buttons) {
+      for (inputSource of frame.session.inputSources) {
+         checkInteraction(button, inputSource, frame, renderer);
+      }
+   }
+}
+
+```
+
+## Gesture detection using this API
+
+One can do gesture detection using the position and orientation values of the various fingers. This can get pretty complicated and stateful, but a straightforward example below would be simplistically detecting that the user has made a fist gesture:
+
+```js
+function checkFistGesture(inputSource, frame, renderer) {
+   for (finger of [[XRJoint.INDEX_PHALANGE_TIP, XRJoint.INDEX_METACARPAL],
+                  [XRJoint.MIDDLE_PHALANGE_TIP, XRJoint.MIDDLE_METACARPAL],
+                  [XRJoint.RING_PHALANGE_TIP, XRJoint.RING_METACARPAL],
+                  [XRJoint.PINKY_PHALANGE_TIP, XRJoint.PINKY_METACARPAL]]) {
+      let tip = finger[0];
+      let metacarpal = finger[1];
+      let tipPose = frame.getPose(inputSource.hand[tip], renderer.referenceSpace);
+      let metacarpalPose = frame.getPose(inputSource.hand[metacarpal], renderer.referenceSpace)
+      if (calculateDistance(tipPose.position, metacarpalPose.position) > minimumDistance ||
+          !checkOrientation(tipPose.orientation, metacarpalPose.orientation)) {
+         return false
+      }
+   }
+   return true;
+}
+
+function checkOrientation(tipOrientation, metacarpalOrientation) {
+   let tipDirection = applyOrientation(tipOrientation, [0, 0, 1]); // Z axis of tip
+   let palmDirection = applyOrientation(metacarpalOrientation, [0, -1, 0]) // -Y axis of metacarpal
+
+   if (1 - dotProduct(tipDirection, palmDirection) < minimumDeviation) {
+      return true;
+   } else {
+      return false;
+   }
+}
+```
+
+## Appendix: Proposed IDL
+
+```webidl
+partial interface XRInputSource {
+   XRHand? hand;
+}
+
+interface XRHand {
+    getter XRJoint(unsigned short jointIndex);
+}
+
+interface XRJoint {
+   XRSpace space;
+   float? radius;
+   
+   const unsigned short WRIST = ..;
+
+   // potentially: const unsigned short THUMB_TRAPEZIUM = ..;
+   const unsigned short THUMB_METACARPAL = ..;
+   const unsigned short THUMB_PHALANGE_PROXIMAL = ..;
+   const unsigned short THUMB_PHALANGE_DISTAL = ..;
+   const unsigned short THUMB_PHALANGE_TIP = ..;
```

Comment:
Including `_PHALANGE` is more specific, though I wonder if it might be overkill, since I believe per-finger anatomy generally only has one thing that is described as "proximal" or "intermediate" or "distal".

Reply:
No strong opinion, I weakly prefer mentioning phalange.

Actually, I just realized that the singular is "phalanx", not "phalange" (cc @Artyom17 , the oculusvr documentation makes the same mistake) so we'd need to change it anyway if we want to keep the PHALANGE.

Reply:
I do think `PHALANX` is a fairly cool word as far as APIs go, though it may be straightforward enough with it, especially for those who don't actually know that this is the singular form of "phalanges".

Reply:
I am _all in_ on making API naming decisions based on the coolness factor. :shipit: